src/bitswap/strategy.rs

```diff
@@ -6,9 +6,9 @@ use std::sync::mpsc::{channel, Sender, Receiver};
 
 pub trait Strategy<TRepoTypes: RepoTypes>: Send + Unpin {
     fn new(repo: Repo<TRepoTypes>) -> Self;
-    fn process_want(&mut self, source: PeerId, cid: Cid, priority: Priority);
-    fn process_block(&mut self, source: PeerId, block: Block);
-    fn poll(&mut self) -> Option<StrategyEvent>;
+    fn process_want(&self, source: PeerId, cid: Cid, priority: Priority);
+    fn process_block(&self, source: PeerId, block: Block);
+    fn poll(&self) -> Option<StrategyEvent>;
 }
```

Comment:
these `&mut`s could likely be added back easily but there was no reason to add them this moment.